### PR TITLE
Add support for parsing index blocks

### PR DESCRIPTION
**Pull request by jpohls1**
Thanks for this great library! This PR adds a function to parse [IndexAllocation](http://inform.pucp.edu.pe/~inf232/Ntfs/ntfs_doc_v0.5/attributes/index_allocation.html) entries to, e.g., navigate through the directory strucuture.

### CODE REVIEW

**Comment by t9t** [comment]
Thanks for the contribution! It has been a while since I played around with NTFS & MFT and worked on this lib, so I need some time to read some docs again and understand what this is about exactly. 😄 

In the mean time, I'm a big supporter of making sure every publicly accessible type and function has well-written documentation. So if that's something that you feel you could do, I would appreciate it if you added that to the pull request.

I'll make sure to come back to this PR soon, after I'm more up to date on the material.

**Comment by jpohls1** [comment]
Thanks for the reply; happy to see you are still maintaining this repository!

>  It has been a while since I played around with NTFS & MFT and worked on this lib, so I need some time to read some docs again and understand what this is about exactly.

Sure thing, NTFS is quite a beast and it takes some time to get that all into the internal RAM (=short term memory). Your library helped me with that quite a lot, though; well-written and clearly structured.

> In the mean time, I'm a big supporter of making sure every publicly accessible type and function has well-written documentation. So if that's something that you feel you could do, I would appreciate it if you added that to the pull request.

That makes sense to me! I just added some comments to the new methods. I am looking forward to your review whenever you have time.

File: mft/mft.go [hunk 1]
```diff
@@ -155,6 +155,16 @@ func applyFixUp(b []byte, offset int, length int) ([]byte, error) {
...
 }
 
+// ApplyFixup applies the NTFS fixup to the data of a Data Run.
+// http://inform.pucp.edu.pe/~inf232/Ntfs/ntfs_doc_v0.5/concepts/fixup.html
+func ApplyFixup(b []byte) ([]byte, error) {
+	r := binutil.NewLittleEndianReader(b)
+	updateSequenceOffset := int(r.Uint16(0x04))
+	updateSequenceSize := int(r.Uint16(0x06))
+	return applyFixUp(b, updateSequenceOffset, updateSequenceSize)
+
```
Comment (t9t) [inline mft/mft.go]: Could you remove this unnecessary new line? 😄

**Comment by t9t** [comment]
I'm glad that people, like you, are actually finding and using this project! 😆 I only made it for a hobby project a few years ago after which I didn't really spend any more time on it.

The added comments look really good, thanks for adding them. 👍 Also very nice that you added the link to documentation, so it was easy to find out more about it.

There's one superfluous empty line, would be neat to remove it, then I'll merge it.

**Comment by jpohls1** [comment]
Sure thing; thanks for the review! I just removed the extra empty line.

**Comment by t9t** [comment]
Great!

**Pull request merged**